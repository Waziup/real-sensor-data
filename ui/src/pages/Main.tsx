import React, { useEffect, useState } from 'react';
import { AppBar, Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, IconButton, makeStyles, Toolbar, Typography } from '@material-ui/core';
import EqualizerIcon from '@material-ui/icons/Equalizer';
import AccountCircleIcon from '@material-ui/icons/AccountCircle';

// import ontologies from "../../data/ontologies/ontologies.json";
import StatusReport from '../layouts/StatusReport';
import SensorSearch from '../layouts/SensorSearch';
import Sensor from '../layouts/Sensor';
import LoginForm from '../layouts/LoginForm';
import UserProfile from '../layouts/UserProfile';
import BackToTop from '../components/BackToTop';

import * as API from "../api";

/*---------------------*/

const useStyles = makeStyles((theme) => ({
  root: {
    '& > *': {
      margin: theme.spacing(1),
      width: '90%',
      display: 'flex'
    },
  },
  button: {
    width: '6rem',
    margin: theme.spacing(1),
  },
  box: {
    border: 'solid 1px #CCC',
    borderRadius: theme.spacing(1),
    margin: theme.spacing(1),
    padding: theme.spacing(1),
    textAlign: "left",
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  fieldset: {
    justifyContent: "center",
    padding: "1rem"
  },
  numInput: {
    width: '5rem',
    marginLeft: '.5rem'
  },
}));

/*---------------------*/

export default function Main() {
  const classes = useStyles();

  /**------------- */

  useEffect(() => {

    handleSearchResultClick({ id: 3494 });

    isAuthorized(); // If user refreshes the page, she/he will be still logged in if was before
    // return () => {}
  }, [])

  /**------------- */

  const [statsDlgOpen, setStatisticsDialogue] = useState(false)
  const handleStatisticsOpen = () => { setStatisticsDialogue(true) }
  const handleStatisticsClose = () => { setStatisticsDialogue(false) }

  /**------------- */

  const [loginDlgOpen, setLoginDlgOpen] = useState(false)
  const handleLoginOpen = () => { setLoginDlgOpen(true) }
  const handleLoginClose = () => { setLoginDlgOpen(false) }

  // If this is not null, it means the user is logged in
  const [user, setUser] = useState(null as API.UserType)
  const loginSuccessCallback = (loggedInUser: API.UserType) => {

    setLoginDlgOpen(false);
    setUser(loggedInUser);
  }

  const logoutSuccessCallback = () => {
    setUser(null);
    // setLoginDlgOpen(false);
  }

  /**------------- */

  const isAuthorized = () => {
    //Just a cheap API call
    API.getUser().then(
      (res) => {
        setUser(res)
      },
      (error) => {
        setUser(null)
      }
    );
  };

  /**------------- */

  const [sensorDlgOpen, setSensorDlgOpen] = useState(false)
  const [sensorProps, setSensorProps] = useState(null)
  const handleSearchResultClick = (dataRow: any) => {
    setSensorDlgOpen(true)
    setSensorProps({ id: dataRow.id })
  }
  const handleSensorDlgClose = () => { setSensorDlgOpen(false) }

  /**------------- */


  return (
    <div className="Main">
      <AppBar position="static">
        <Toolbar>
          {/* <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
            <MenuIcon />
          </IconButton> */}
          <Typography variant="h6" className={classes.title}>
            Sensor Data Simulator
           </Typography>

          <IconButton
            edge="end"
            aria-label="Statistics"
            aria-haspopup="true"
            onClick={handleStatisticsOpen}
            color="inherit"
          >
            <EqualizerIcon />
          </IconButton>

          <IconButton
            edge="end"
            aria-label="login"
            aria-haspopup="true"
            onClick={handleLoginOpen}
            color="inherit"
          >
            <AccountCircleIcon />
          </IconButton>

        </Toolbar>
      </AppBar>

      {/* ------------------------- */}

      <Toolbar id="back-to-top-anchor" />

      <SensorSearch onSearchResultClick={handleSearchResultClick} />

      <BackToTop topAnchorId="back-to-top-anchor" />

      {/* ------------------------- */}

      <Dialog
        open={sensorDlgOpen}
        onClose={handleSensorDlgClose}
        aria-labelledby="sensor-dialog-title"
        aria-describedby="sensor-dialog-description"
        fullScreen={true}
      >
        <DialogTitle id="sensor-dialog-title">{"Sensor Details"}</DialogTitle>
        <DialogContent>
          {sensorDlgOpen && <Sensor {...sensorProps} />}
        </DialogContent>
        <DialogActions>
          <Button onClick={handleSensorDlgClose} color="primary" autoFocus>
            Close
          </Button>
        </DialogActions>
      </Dialog>

      {/* ------------------------- */}

      <Dialog
        open={statsDlgOpen}
        onClose={handleStatisticsClose}
        aria-labelledby="statistics-dialog-title"
        aria-describedby="statistics-dialog-description"
        maxWidth={false}
      >
        <DialogTitle id="statistics-dialog-title">{"Data Collection Statistics"}</DialogTitle>
        <DialogContent >
          <StatusReport />
        </DialogContent>
        <DialogActions>
          <Button onClick={handleStatisticsClose} color="primary" autoFocus>
            Close
          </Button>
        </DialogActions>
      </Dialog>

      {/* ------------------------- */}

      <Dialog
        open={loginDlgOpen}
        onClose={handleLoginClose}
        aria-labelledby="login-dialog-title"
        aria-describedby="login-dialog-description"
        maxWidth={false}
      >
        <DialogContent >
          {user !== null ?
            <UserProfile user={user} onLogout={logoutSuccessCallback} /> :
            <LoginForm onSuccess={loginSuccessCallback} />}
        </DialogContent>
        <DialogActions>
          <Button onClick={handleLoginClose} color="primary" autoFocus>
            Close
          </Button>
        </DialogActions>
      </Dialog>

    </div >
  );
}

/**---------------- */