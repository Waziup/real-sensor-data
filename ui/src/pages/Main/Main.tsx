import React, { useEffect, useState } from 'react';
import { AppBar, Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, IconButton, makeStyles, Toolbar, Typography } from '@material-ui/core';
import EqualizerIcon from '@material-ui/icons/Equalizer';

// import ontologies from "../../data/ontologies/ontologies.json";
import StatusReport from '../../components/StatusReport';
import SensorSearch from '../../components/SensorSearch';

/*---------------------*/

const useStyles = makeStyles((theme) => ({
  root: {
    '& > *': {
      margin: theme.spacing(1),
      width: '90%',
      display: 'flex'
    },
  },
  button: {
    width: '6rem',
    margin: theme.spacing(1),
  },
  box: {
    border: 'solid 1px #CCC',
    borderRadius: theme.spacing(1),
    margin: theme.spacing(1),
    padding: theme.spacing(1),
    textAlign: "left",
  },
  menuButton: {
    marginRight: theme.spacing(2),
  },
  title: {
    flexGrow: 1,
  },
  fieldset: {
    justifyContent: "center",
    padding: "1rem"
  },
  numInput: {
    width: '5rem',
    marginLeft: '.5rem'
  },
}));

/*---------------------*/

export default function Main() {
  const classes = useStyles();

  /**------------- */

  const [statsDlgOpen, setStatisticsDialogue] = useState(false)
  const handleStatisticsOpen = () => { setStatisticsDialogue(true) }
  const handleStatisticsClose = () => { setStatisticsDialogue(false) }

  /**------------- */


  return (
    <div className="Main">
      <AppBar position="static">
        <Toolbar>
          {/* <IconButton edge="start" className={classes.menuButton} color="inherit" aria-label="menu">
            <MenuIcon />
          </IconButton> */}
          <Typography variant="h6" className={classes.title}>
            Sensor Data Simulator
           </Typography>
          {/* <Button color="inherit">Login</Button> */}
          <IconButton
            edge="end"
            aria-label="Statistics"
            aria-haspopup="true"
            onClick={handleStatisticsOpen}
            color="inherit"
          >
            <EqualizerIcon />
          </IconButton>
        </Toolbar>
      </AppBar>

      <SensorSearch />


      <Dialog
        open={statsDlgOpen}
        onClose={handleStatisticsClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">{"Data Collection Statistics"}</DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">
            <StatusReport />
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button onClick={handleStatisticsClose} color="primary" autoFocus>
            Close
          </Button>
        </DialogActions>
      </Dialog>

    </div >
  );
}

/**---------------- */