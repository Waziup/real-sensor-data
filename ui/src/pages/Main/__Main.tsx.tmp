import React, { useEffect, useState } from 'react';

import { Button, Container, FormControl, FormGroup, FormLabel, makeStyles, TextField } from '@material-ui/core';

import LinearProgressWithLabel from "../../components/LinearProgressWithLabel"

import ontologies from "../../data/ontologies/ontologies.json";

/*---------------------*/

const useStyles = makeStyles((theme) => ({
  root: {
    '& > *': {
      margin: theme.spacing(1),
      width: '90%',
      display: 'flex'
    },
  },
  button: {
    width: '6rem',
    margin: theme.spacing(1),
  },
  box: {
    border: 'solid 1px #CCC',
    borderRadius: theme.spacing(1),
    margin: theme.spacing(1),
    padding: theme.spacing(1),
    textAlign: "left",
  },
  fieldset: {
    justifyContent: "center",
    padding: "1rem"
  },
  numInput: {
    width: '5rem',
    marginLeft: '.5rem'
  },
}));

/*---------------------*/

type formValuesType = {
  numDevices: number;
  minSensors: number;
  maxSensors: number;
  minValues: number;
  maxValues: number;
  minEntries: number;
  maxEntries: number;
  minTime: number;
  maxTime: number;
};


type SensorMetaType = {
  "kind"?: string;
  "quantity"?: string;
  "unit"?: string;
};

type SensorType = {
  "name": string;
  "meta"?: SensorMetaType;
};

type DeviceMetaType = {
  "codec"?: string;
};

type DeviceType = {
  "name": string;
  "sensors"?: SensorType[];
  "meta"?: DeviceMetaType;
}

/*---------------------*/

export default function Main() {
  const classes = useStyles();

  const [formValues, setFormValues] = useState({
    numDevices: 1,
    minSensors: 1,
    maxSensors: 5,
    minValues: -10,
    maxValues: 100,
    minEntries: 5,
    maxEntries: 10,
    minTime: 10,
    maxTime: 30,
  } as formValuesType);

  const [selectedCodec, setSelectedCodec] = useState(null);
  const [progress, setProgress] = useState(0);

  const startDataGenerator = async () => {

    // var mu = [0, 0],
    //   sigma = [0.25, 0.5],
    //   correlation = [[1.0, 0.5], [0.5, 1.0]];

    // var data = generateCorrLognorm(100, mu, sigma, correlation);

    // return;

    for (let d = 0; d < formValues.numDevices; d++) {
      setProgress(Math.round(100 * d / formValues.numDevices));

      const deviceName = getRandomDeviceName();
      const numOfSensors = getRandomValue(formValues.minSensors, formValues.maxSensors);

      let deviceObj: DeviceType = {
        name: deviceName,
        sensors: [],
        meta: {
          codec: selectedCodec
        }
      };

      for (let s = 0; s < numOfSensors; s++) {
        deviceObj?.sensors?.push(getRandomSensor());
      }

      // Pushing devices data

      let deviceID = 0;
      try {
        deviceID = await postAPI('/devices', deviceObj);
        if (!deviceID) continue;
      } catch (e) {
        console.error(e);
        continue;
      }

      //Preparing the time delays

      const numOfEntries = getRandomValue(formValues.minEntries, formValues.maxEntries);
      let timeSet = [];
      let nowObj = new Date();
      // Let's enter the values in the near past
      nowObj.setMinutes(nowObj.getMinutes() - formValues.maxTime * numOfEntries);
      for (let e = 0; e < numOfEntries; e++) {
        const delayInMinutes = getRandomValue(formValues.minTime, formValues.maxTime);
        nowObj.setMinutes(nowObj.getMinutes() + delayInMinutes);
        timeSet.push(nowObj.toISOString());
      }

      //Pushing the sensor values
      const sensorsList = await getAPI(`/devices/${deviceID}/sensors`)
      for (let sensor of sensorsList) {

        const sensorID = sensor['id'];

        const randomValues = getCorrRdandomValues(numOfEntries, formValues.minValues, formValues.maxValues);
        let sensorValues = [];
        for (let e = 0; e < numOfEntries; e++) {
          const value = randomValues[e];
          const time = timeSet[e];
          sensorValues.push({ value, time });
        }

        try {
          await postAPI(`/devices/${deviceID}/sensors/${sensorID}/values`, sensorValues);
        } catch (e) {
          console.error(e);
          continue;
        }
      }

    }
    setProgress(100);
  }

  /**------------- */

  const stopDataGenerator = () => { }

  /**------------- */


  return (
    <div className="Main">
      <Container maxWidth="md">
        <form className={classes.root} noValidate autoComplete="on" onSubmit={e => e.preventDefault()}>
          <FormControl className={classes.box}>
            <FormLabel>Devices:</FormLabel>
            <FormGroup className={classes.fieldset}>
              <TextField
                id="numDevices"
                label="Number of devices"
                value={formValues.numDevices}
                type="number"
                onChange={(e) => setFormValues({ ...formValues, numDevices: Number(e.target.value) })}
              />
              <Codecs onChange={(codec: string) => { setSelectedCodec(codec); }}></Codecs>
            </FormGroup>
          </FormControl>

          <FormControl className={classes.box}>
            <FormLabel>Num of Sensors:</FormLabel>
            <FormGroup row className={classes.fieldset}>
              <TextField id="minSensors" className={classes.numInput} label="Min" value={formValues.minSensors} type="number" onChange={(e) => setFormValues({ ...formValues, minSensors: Number(e.target.value) })} />
              <TextField id="maxSensors" className={classes.numInput} label="Max" value={formValues.maxSensors} type="number" onChange={(e) => setFormValues({ ...formValues, maxSensors: Number(e.target.value) })} />
            </FormGroup>
          </FormControl>

          <FormControl className={classes.box}>
            <FormLabel>Values Range:</FormLabel>
            <FormGroup row className={classes.fieldset}>
              <TextField id="minValues" className={classes.numInput} label="Min" value={formValues.minValues} type="number" onChange={(e) => setFormValues({ ...formValues, minValues: Number(e.target.value) })} />
              <TextField id="maxValues" className={classes.numInput} label="Max" value={formValues.maxValues} type="number" onChange={(e) => setFormValues({ ...formValues, maxValues: Number(e.target.value) })} />
            </FormGroup>
          </FormControl>

          <FormControl className={classes.box}>
            <FormLabel>Num of value entries for each device:</FormLabel>
            <FormGroup row className={classes.fieldset}>
              <TextField id="minEntries" className={classes.numInput} label="Min" value={formValues.minEntries} type="number" onChange={(e) => setFormValues({ ...formValues, minEntries: Number(e.target.value) })} />
              <TextField id="maxEntries" className={classes.numInput} label="Max" value={formValues.maxEntries} type="number" onChange={(e) => setFormValues({ ...formValues, maxEntries: Number(e.target.value) })} />
            </FormGroup>
          </FormControl>

          <FormControl className={classes.box}>
            <FormLabel>Delay between two entries (in minutes):</FormLabel>
            <FormGroup row className={classes.fieldset}>
              <TextField id="minTime" className={classes.numInput} label="Min" value={formValues.minTime} type="number" onChange={(e) => setFormValues({ ...formValues, minTime: Number(e.target.value) })} />
              <TextField id="maxTime" className={classes.numInput} label="Max" value={formValues.maxTime} type="number" onChange={(e) => setFormValues({ ...formValues, maxTime: Number(e.target.value) })} />
            </FormGroup>
          </FormControl>

          <FormGroup row style={{ justifyContent: "center" }}>
            {progress > 0 && <LinearProgressWithLabel value={progress} />}
            <Button variant="contained" disabled={progress > 0 && progress < 100} className={classes.button} onClick={startDataGenerator} type="submit" color="primary">Start</Button>
            {/* <Button variant="contained" className={classes.button} onClick={stopDataGenerator} color="secondary">Stop</Button> */}
          </FormGroup>
        </form>
      </Container>
    </div >
  );
}

/**---------------- */

export const getRandomSensor = (): SensorType => {

  const sensingDevicesKeys = Object.keys(ontologies.sensingDevices);
  const randomKindId = Math.floor(Math.random() * sensingDevicesKeys.length);

  const randomKind = sensingDevicesKeys[randomKindId];
  const randomSensor = ontologies.sensingDevices[sensingDevicesKeys[randomKindId]];

  const randomQuantity = randomSensor.quantities.length ? randomSensor.quantities[Math.floor(Math.random() * randomSensor.quantities.length)] : "";
  const randomUnitId = randomQuantity ? Math.floor(Math.random() * ontologies.quantities[randomQuantity].units.length) : null;
  const randomUnit = randomUnitId !== null ? ontologies.quantities[randomQuantity].units[randomUnitId] : "";

  return {
    "name": randomSensor.label,
    "meta": {
      "kind": randomKind,
      "quantity": randomQuantity,
      "unit": randomUnit,
    }
  }
}

/**------------- */

export const getRandomDeviceName = (): string => {
  const randomDeviceId = Math.floor(Math.random() * deviceNames.length);
  return deviceNames[randomDeviceId];
}

/**------------- */

export const getRandomValue = (min: number, max: number) => {
  return Math.round(Math.random() * (max - min) + min);
}

/**------------- */

export const getCorrRdandomValues = (count: number, min: number, max: number) => {

  let output: number[] = [];
  output.push(Math.round(Math.random() * (max - min) + min));
  for (let i = 1; i < count; i++) {
    let localMin = output[i - 1] - (max - min) * 0.05; // Near numbers with max 5% distance
    let localMax = output[i - 1] + (max - min) * 0.05;
    localMin = localMin < min ? min : localMin;
    localMax = localMax > max ? max : localMax;
    const newNumber = Math.random() * (localMax - localMin) + localMin;
    output.push(Math.round(newNumber));
  }
  // console.log(count, output);
  return output;
}

/**------------- */

export const postAPI = async (path: string, val: any) => {
  let resp = await fetch(path, {
    method: "POST",
    headers: {
      "Content-Type": "application/json; charset=utf-8"
    },
    body: JSON.stringify(val),
  });
  const contentType = resp.headers.get("Content-Type");
  const isJson = contentType?.startsWith("application/json");
  if (!resp.ok) {
    if (isJson) {
      var data = await resp.json();
      throw `HTTP Error ${resp.status} ${resp.statusText}\n${data}`;
    } else {
      var text = await resp.text();
      throw `HTTP Error ${resp.status} ${resp.statusText}\n${data}`;
    }
  }
  return isJson ? resp.json() : resp.text();
}

/**------------- */

export const getAPI = async (path: string) => {
  let resp = await fetch(path, {
    headers: {
      "Content-Type": "application/json; charset=utf-8"
    }
  });
  const contentType = resp.headers.get("Content-Type");
  const isJson = contentType?.startsWith("application/json");
  if (!resp.ok) {
    if (isJson) {
      var data = await resp.json();
      throw `HTTP Error ${resp.status} ${resp.statusText}\n${data}`;
    } else {
      var text = await resp.text();
      throw `HTTP Error ${resp.status} ${resp.statusText}\n${data}`;
    }
  }
  return isJson ? resp.json() : resp.text();
}

/**------------- */