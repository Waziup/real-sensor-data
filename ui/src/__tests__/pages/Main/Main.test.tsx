import React from 'react';
import { create } from 'react-test-renderer';
import Main from '../../../pages/Main/Main';


describe("Main page", () => {
  it("renders correctly", () => {
    const tree = create(<Main />);
    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("Genrates a random device name correctly", () => {

    const Main = require('../../../pages/Main/Main');

    // const mock = jest.spyOn(Main, "getRandomDeviceName");
    expect(Main.getRandomDeviceName()).toBeTruthy();
    expect(Main.getRandomDeviceName().length).not.toEqual(0);
    expect(typeof Main.getRandomDeviceName()).toBe("string");
  });

  it("Genrates a random Sensor correctly", () => {

    const Main = require('../../../pages/Main/Main');

    expect(Main.getRandomSensor()).toBeTruthy();
    expect(typeof Main.getRandomSensor()).toBe("object");

  });

  it("Genrates a random Value correctly", () => {

    const Main = require('../../../pages/Main/Main');

    expect(Main.getRandomValue(0, 1000)).toBeTruthy();
    expect(typeof Main.getRandomValue(-10, 1000)).toBe("number");

  });


});