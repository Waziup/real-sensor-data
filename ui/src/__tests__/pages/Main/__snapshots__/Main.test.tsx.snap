// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page renders correctly 1`] = `
<div
  className="Main"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <form
      autoComplete="on"
      className="makeStyles-root-1"
      noValidate={true}
      onSubmit={[Function]}
    >
      <div
        className="MuiFormControl-root makeStyles-box-3"
      >
        <label
          className="MuiFormLabel-root"
        >
          Devices:
        </label>
        <div
          className="MuiFormGroup-root makeStyles-fieldset-4"
        >
          <div
            className="MuiFormControl-root MuiTextField-root"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="numDevices"
              id="numDevices-label"
            >
              Number of devices
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="numDevices"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={10}
              />
            </div>
          </div>
          <div>
            Loading codecs...
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root makeStyles-box-3"
      >
        <label
          className="MuiFormLabel-root"
        >
          Num of Sensors:
        </label>
        <div
          className="MuiFormGroup-root makeStyles-fieldset-4 MuiFormGroup-row"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-numInput-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="minSensors"
              id="minSensors-label"
            >
              Min
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="minSensors"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={1}
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-numInput-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="maxSensors"
              id="maxSensors-label"
            >
              Max
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="maxSensors"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={5}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root makeStyles-box-3"
      >
        <label
          className="MuiFormLabel-root"
        >
          Values Range:
        </label>
        <div
          className="MuiFormGroup-root makeStyles-fieldset-4 MuiFormGroup-row"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-numInput-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="minValues"
              id="minValues-label"
            >
              Min
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="minValues"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={-1000}
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-numInput-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="maxValues"
              id="maxValues-label"
            >
              Max
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="maxValues"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={1000}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root makeStyles-box-3"
      >
        <label
          className="MuiFormLabel-root"
        >
          Num of value entries for each device:
        </label>
        <div
          className="MuiFormGroup-root makeStyles-fieldset-4 MuiFormGroup-row"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-numInput-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="minEntries"
              id="minEntries-label"
            >
              Min
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="minEntries"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={5}
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-numInput-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="maxEntries"
              id="maxEntries-label"
            >
              Max
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="maxEntries"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={10}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiFormControl-root makeStyles-box-3"
      >
        <label
          className="MuiFormLabel-root"
        >
          Delay between two entries (in minutes):
        </label>
        <div
          className="MuiFormGroup-root makeStyles-fieldset-4 MuiFormGroup-row"
        >
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-numInput-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="minTime"
              id="minTime-label"
            >
              Min
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="minTime"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={10}
              />
            </div>
          </div>
          <div
            className="MuiFormControl-root MuiTextField-root makeStyles-numInput-5"
          >
            <label
              className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled"
              data-shrink={true}
              htmlFor="maxTime"
              id="maxTime-label"
            >
              Max
            </label>
            <div
              className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-formControl MuiInput-formControl"
              onClick={[Function]}
            >
              <input
                aria-invalid={false}
                autoFocus={false}
                className="MuiInputBase-input MuiInput-input"
                disabled={false}
                id="maxTime"
                onAnimationStart={[Function]}
                onBlur={[Function]}
                onChange={[Function]}
                onFocus={[Function]}
                required={false}
                type="number"
                value={30}
              />
            </div>
          </div>
        </div>
      </div>
      <div
        className="MuiFormGroup-root MuiFormGroup-row"
        style={
          Object {
            "justifyContent": "center",
          }
        }
      >
        <button
          className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-2 MuiButton-containedPrimary"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="submit"
        >
          <span
            className="MuiButton-label"
          >
            Start
          </span>
        </button>
      </div>
    </form>
  </div>
</div>
`;
